core/src/main/java/com/sequenceiq/cloudbreak/service/upgrade/UpgradePreconditionService.java

```diff
@@ -58,7 +59,8 @@ private String getNotUpgradeableAttachedClusters(List<? extends StackDtoDelegate
                 .stream()
                 .filter(stackDto -> BlueprintUpgradeOption.GA != Optional.ofNullable(stackDto.getBlueprint())
                         .map(Blueprint::getBlueprintUpgradeOption)
-                        .orElse(null))
+                        .orElse(null) && ResourceStatus.USER_MANAGED != Optional.ofNullable(stackDto.getBlueprint())
```

Comment:
I don't think this is sufficient. This will make sure that the DL itself can be upgrade if there are Data Hubs with custom templates, but then the DH upgrade will still be rejected. If you take a look at this method, https://github.com/hortonworks/cloudbreak/blob/e0bd28ed3340d9e8ee3b2b9aa7315377b7e3371d/core-model/src/main/java/com/sequenceiq/cloudbreak/domain/BlueprintUpgradeOption.java#L48 I think this will still reject the upgrade for a Data hub with custom template.

Reply:
@keyki for standalone DH upgrade (DH upgrade without DL) the BluePrintUpgradeOption - is not being checked at all in code. @nadamdb pointed it out and I went through the code completely.